src/transformStateWithClones.js

```diff
@@ -7,7 +7,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ADD_PROPERTIES = 'addProperties';
+  const REMOVE_PROPERTIES = 'removeProperties';
+  const CLEAR = 'clear';
+
+  const stateCopy = { ...state };
+  const resultObject = [];
+
+  for (const item of actions) {
+    switch (item.type) {
+      case ADD_PROPERTIES: {
+        Object.assign(stateCopy, item.extraData);
+        resultObject.push({ ...stateCopy });
+        break;
+      }
+
+      case REMOVE_PROPERTIES: {
+        for (const key of item.keysToRemove) {
+          delete stateCopy[key];
+        }
+        resultObject.push({ ...stateCopy });
+        break;
+      }
+
+      case CLEAR: {
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        resultObject.push({});
```

Comment:
I would move **resultObject.push({ ...stateCopy });** outside the switch. You don't need to declare it separately in each type of action. If the actions are done correctly, then just having **resultObject.push({ ...stateCopy });** outside of switch would have the same effect with it only being written once instead of thrice.